typescript/sdk/src/fee/EvmTokenFeeModule.ts

```diff
@@ -141,23 +181,117 @@ export class EvmTokenFeeModule extends HyperlaneModule<
-  ): Promise<
-    Annotated<ProtocolTypedTransaction<ProtocolType.Ethereum>['transaction']>[]
-  > {
-    throw new Error('Not implemented');
+    targetConfig: TokenFeeConfigInput,
+    params?: Partial<ReaderParams>,
+  ): Promise<AnnotatedEV5Transaction[]> {
+    let updateTransactions: AnnotatedEV5Transaction[] = [];
+
+    const normalizedTargetConfig: TokenFeeConfig = normalizeConfig(
+      await EvmTokenFeeModule.expandConfig({
+        config: targetConfig,
+        multiProvider: this.multiProvider,
+        chainName: this.chainName,
+      }),
+    );
+
+    const actualConfig = await this.read(params);
+    const normalizedActualConfig: TokenFeeConfig =
+      normalizeConfig(actualConfig);
+
+    //If configs are the same, return empty array
+    if (deepEquals(normalizedActualConfig, normalizedTargetConfig)) {
+      this.logger.debug(
+        `Same config for ${normalizedTargetConfig.type}, no update needed`,
+      );
+      return [];
+    }
+
+    // if the type is a immutable type, then redeploy
+    if (
+      ImmutableTokenFeeType.includes(
+        normalizedTargetConfig.type as (typeof ImmutableTokenFeeType)[number],
+      )
+    ) {
+      this.logger.info(
+        `Immutable fee type ${normalizedTargetConfig.type}, redeploying`,
+      );
+      const contracts = await this.deploy({
+        config: normalizedTargetConfig,
+        multiProvider: this.multiProvider,
+        chainName: this.chainName,
+        contractVerifier: this.contractVerifier,
+      });
+      this.args.addresses.deployedFee =
+        contracts[this.chainName][normalizedTargetConfig.type].address;
+    }
+
+    // if the type is a mutable type, then update
+    updateTransactions = [
+      ...(await this.updateRoutingFee(
+        _.merge(
+          actualConfig,
+          normalizedTargetConfig,
+        ) as DerivedRoutingFeeConfig,
+      )),
+      ...this.createOwnershipUpdateTxs(
+        normalizedActualConfig,
+        normalizedTargetConfig,
+      ),
+    ];
+
+    return updateTransactions;
+  }
+
+  private async updateRoutingFee(targetConfig: DerivedRoutingFeeConfig) {
+    const updateTransactions: AnnotatedEV5Transaction[] = [];
+
+    if (!targetConfig.feeContracts) return [];
+    const currentRoutingAddress = this.args.addresses.deployedFee;
+    await promiseObjAll(
+      objMap(targetConfig.feeContracts, async (chainName, config) => {
+        const address = config.address;
+        await this.update(config, { address });
+
+        // fetches the latest deployedFee
+        updateTransactions.push({
+          annotation: 'Updating routing fee...',
+          chainId: this.chainId,
+          to: currentRoutingAddress,
+          data: RoutingFee__factory.createInterface().encodeFunctionData(
+            'setFeeContract(uint32,address)',
+            [
+              this.multiProvider.getDomainId(chainName),
+              this.args.addresses.deployedFee,
+            ],
+          ),
+        });
```

Comment:
this tells me that a batch method to set multiple fee contracts in batch would be helpful